crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -134,6 +133,13 @@ where
                 )
             })
             .collect::<Result<Vec<_>, CircuitError>>()?;
+        // Adding the last if STAKE_TABLE_CAPACITY is not a multiple of 2
+        if STAKE_TABLE_CAPACITY % 2 == 1 {
+            signed_amount_var.push(circuit.mul(
+                stake_table_var[STAKE_TABLE_CAPACITY - 1].stake_amount,
+                signer_bit_vec_var[STAKE_TABLE_CAPACITY - 1].0,
+            )?);
+        }
```

Comment:
wait, wouldn't this cause double counting the last pk's vote twice (if he voted)?

Reply:
No, for example if `STAKE_TABLE_CAPACITY` is 3, then previous computation only takes first 2 indices, and this one takes the last.

Reply:
ah, make sense!